calicoctl/commands/version.go

```diff
@@ -130,3 +130,79 @@ Description:
 
 	return err
 }
+
+func GetConfigAndContext(args []string) (string, string) {
+	// Possible arg formats:
+	// -cVALUE, -c VALUE, --config VALUE, --config=VALUE,
+	// --context VALUE, --context=VALUE
+	var config, context string
+
+	for i, arg := range args {
```

Comment:
You might've tried this, but I'm wondering instead of doing all the parsing manually which opens us up to a lot of edge cases if we could add a fake docopt string that displays all the options across all the commands (to avoid errors from unknown options) so we could catch the `--config` option. Of course this also has the downside of needing to be updated if we add new calicoctl commands with different options but it does give us a fairly effortless way of getting the token parsing for free (without pulling relevant code from docopt). WDYT?

Reply:
hm... tbh it didn't cross my mind to have every single argument possible in the docstring, I'm gonna try that. thanks!